mantidproject/mantid
# Get ConvertWANDSCDtoQ and FindPeaksMD working for HB3A

**Description of work.**

This make the algorithms ConvertWANDSCDtoQ and FindPeaksMD work with data from DEMAND (HB3A). It's mainly fixing the handling of the goniometers.

**To test:**

This load some data, converts it to `Q_sample` and `HKL` and find peaks from the `Q_sample` workspace. You should check that the HKL Frame has the peaks aligning on integer HKL's and that the peaks workspace has correct wavelength and sensible HKL's.

```python
from mantid.simpleapi import *

data = Load("/HFIR/HB3A/IPTS-24655/shared/autoreduce/HB3A_exp0724_scan0183.nxs")
SetGoniometer(data, Axis0='omega,0,1,0,-1',Axis1='chi,0,0,1,-1',Axis2='phi,0,1,0,-1')
ConvertWANDSCDtoQ(InputWorkspace=data,
                  NormaliseBy='None',
                  Wavelength=1.008,
                  BinningDim0='-8.02,8.02,401',
                  BinningDim1='-4.02,4.02,201',
                  BinningDim2='-8.02,8.02,401',
                  KeepTemporaryWorkspaces=True,
                  OutputWorkspace='Q')

FindPeaksMD(InputWorkspace='Q_data',
            PeakDistanceThreshold=0.25,
            DensityThresholdFactor=5000,
            OutputWorkspace='peaks',
            CalculateGoniometerForCW=True,
            Wavelength=1.008,
            FlipX=True, 
            InnerGoniometer=True)
IndexPeaks('peaks')

ConvertWANDSCDtoQ(InputWorkspace=data,
                  NormaliseBy='None',
                  Wavelength=1.008,
                  Frame='HKL',
                  BinningDim0='-2.02,5.02,176',
                  BinningDim1='-2.02,5.02,176',
                  BinningDim2='-16.04,16.04,401',
                  OutputWorkspace='HKL')
```

![peaks](https://user-images.githubusercontent.com/5595210/70156638-40fb6c00-1682-11ea-85a3-88e1ee4a7e99.png)

---

#### Reviewer ####

Please comment on the following ([full description](http://developer.mantidproject.org/ReviewingAPullRequest.html)):

##### Code Review #####

- Is the code of an acceptable quality?
- Does the code conform to the [coding standards](http://developer.mantidproject.org/Standards/)?
- Are the unit tests small and test the class in isolation?
- If there are changes in the release notes then do they describe the changes appropriately?

##### Functional Tests #####

- Do changes function as described? Add comments below that describe the tests performed?
- Do the changes handle unexpected situations, e.g. bad input?
- Has the relevant (user and developer) documentation been added/updated?

Does everything look good? Mark the review as **Approve**. A member of `@mantidproject/gatekeepers` will take care of it.
